File: frontend/src/api/certificateAlert.ts
import client from './client';
import type { ApiResponse } from '../types';

export interface CertificateAlertConfig {
  id: number;
  certificate_id: number;
  enabled: boolean;
  threshold_days: string;
  notify_internal: boolean;
  notify_email: boolean;
  notify_webhook: boolean;
  email_recipients: string;
  created_at: string;
  updated_at: string;
}

export interface AlertConfigRequest {
  certificate_id: number;
  enabled: boolean;
  threshold_days: string;
  notify_internal: boolean;
  notify_email: boolean;
  notify_webhook: boolean;
  email_recipients: string;
}

export const certificateAlertAPI = {
  createOrUpdate: async (data: AlertConfigRequest) => {
    const response = await client.post<ApiResponse<CertificateAlertConfig>>('/certificate-alerts', data);
    return (response as unknown as ApiResponse<CertificateAlertConfig>).data!;
  },

  get: async (certificateId: number) => {
    const response = await client.get<ApiResponse<CertificateAlertConfig>>(`/certificate-alerts/${certificateId}`);
    return (response as unknown as ApiResponse<CertificateAlertConfig>).data!;
  },
};

File: backend/internal/models/certificate_alert.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type CertificateAlertConfig struct {
	ID             uint           `gorm:"primarykey" json:"id"`
	CreatedAt      time.Time      `json:"created_at"`
	UpdatedAt      time.Time      `json:"updated_at"`
	DeletedAt      gorm.DeletedAt `gorm:"index" json:"-"`
	CertificateID  uint           `gorm:"uniqueIndex;not null" json:"certificate_id"`
	Enabled        bool           `gorm:"default:true" json:"enabled"`
	ThresholdDays  string         `gorm:"type:text" json:"threshold_days"` // JSON array: [30,7,1]
	NotifyInternal bool           `gorm:"default:true" json:"notify_internal"`
	NotifyEmail    bool           `gorm:"default:false" json:"notify_email"`
	NotifyWebhook  bool           `gorm:"default:false" json:"notify_webhook"`
	EmailRecipients string        `gorm:"type:text" json:"email_recipients"` // JSON array
}

type CertificateAlertLog struct {
	ID            uint      `gorm:"primarykey" json:"id"`
	CreatedAt     time.Time `json:"created_at"`
	CertificateID uint      `gorm:"not null;index:idx_cert_alert_dedup" json:"certificate_id"`
	ThresholdDays int       `gorm:"not null;index:idx_cert_alert_dedup" json:"threshold_days"`
	AlertDate     string    `gorm:"type:varchar(10);not null;index:idx_cert_alert_dedup" json:"alert_date"` // YYYY-MM-DD
	NotifiedAt    time.Time `json:"notified_at"`
}
